L1Trigger/TrackFindingTracklet/src/TETableInner.cc

```diff
@@ -0,0 +1,311 @@
+#include "L1Trigger/TrackFindingTracklet/interface/TETableInner.h"
+
+using namespace std;
+using namespace Trklet;
+
+TETableInner::TETableInner(const Settings* settings) : TETableBase(settings) { nbits_ = 20; }
+
+TETableInner::TETableInner(
+    const Settings* settings, int layer1, int layer2, int layer3, int zbits, int rbits, bool thirdLayerIsDisk)
+    : TETableBase(settings) {
+  nbits_ = 20;
+  init(settings, layer1, layer2, layer3, zbits, rbits, thirdLayerIsDisk);
+}
+
+void TETableInner::init(const Settings* settings, int layer1, int layer2, int zbits, int rbits) {
+  init(settings, layer1, layer2, -1, zbits, rbits);
+}
+
+void TETableInner::init(
+    const Settings* settings, int layer1, int layer2, int layer3, int zbits, int rbits, bool thirdLayerIsDisk) {
+  thirdLayerIsDisk_ = thirdLayerIsDisk;
+
+  layer1_ = layer1;
+  layer2_ = layer2;
+  layer3_ = layer3;
+  zbits_ = zbits;
+  rbits_ = rbits;
+
+  bool extended = layer1 == 2 && layer2 == 3 && layer3_ == 1 && thirdLayerIsDisk_;
+  bool extra = layer1 == 2 && layer2 == 3 && !extended;
+
+  rbins_ = (1 << rbits);
+  rminl1_ = settings->rmean(layer1 - 1) - settings->drmax();
+  rmaxl1_ = settings->rmean(layer1 - 1) + settings->drmax();
+  dr_ = 2 * settings->drmax() / rbins_;
+
+  zbins_ = (1 << zbits);
+  zminl1_ = -settings->zlength();
+  zminl2_ = settings->zlength();
+  dz_ = 2 * settings->zlength() / zbins_;
+
+  if (layer1 == 1) {
+    rmindisk_ = settings->rmindiskvm();
+    rmaxdisk_ = settings->rmaxdiskl1overlapvm();
+  }
+
+  if (layer1 == 2) {
+    rmindisk_ = settings->rmindiskl2overlapvm();
+    rmaxdisk_ = (extended ? settings->rmaxdisk() : settings->rmaxdiskvm());
+  }
+
+  rmeanl2_ = settings->rmean(layer2 - 1);
+  if (layer3_ > 0) {
+    rmeanl3_ = settings->rmean(layer3 - 1);
+    zmeand3_ = settings->zmean(layer3 - 1);
+  } else {
+    rmeanl3_ = 0.;
+    zmeand3_ = 0.;
+  }
+
+  for (int izbin = 0; izbin < zbins_; izbin++) {
+    for (int irbin = 0; irbin < rbins_; irbin++) {
+      int value = getLookupValue(settings, izbin, irbin, extra);
+      table_.push_back(value);
+    }
+  }
+
+  if (settings->writeTable()) {
+    writeVMTable("VMTableInnerL" + std::to_string(layer1_) + "L" + std::to_string(layer2_) + ".tab");
+  }
+}
+
+int TETableInner::getLookupValue(const Settings* settings, int izbin, int irbin, bool extra) {
+  double z1 = zminl1_ + izbin * dz_;
+  double z2 = zminl1_ + (izbin + 1) * dz_;
+
+  double r1 = rminl1_ + irbin * dr_;
+  double r2 = rminl1_ + (irbin + 1) * dr_;
+
+  if (extra and
+      std::abs(0.5 * (z1 + z2)) < 52.0) {  //This seeding combinations should not be for central region of detector
```

Comment:
comment on separate line

Reply:
class entirely deleted as part of additional code simplifications